sdk/python/tests/integration/feature_repos/universal/data_sources/file.py

```diff
@@ -352,3 +357,56 @@ def create_offline_store_config(self):
             staging_location_endpoint_override=self.endpoint_url,
         )
         return self.duckdb_offline_store_config
+
+
+class RemoteOfflineStoreDataSourceCreator(FileDataSourceCreator):
+    def __init__(self, project_name: str, *args, **kwargs):
+        super().__init__(project_name)
+        self.server_port: int = 0
+        self.proc: Process = None
+
+    def setup(self, registry: RegistryConfig):
+        parent_offline_config = super().create_offline_store_config()
+
+        fs = FeatureStore(
+            config=RepoConfig(
+                project=self.project_name,
+                provider="local",
+                offline_store=parent_offline_config,
+                registry=registry.path,
+                entity_key_serialization_version=2,
+            )
+        )
+        self.server_port = free_port()
+        host = "0.0.0.0"
+        self.proc = Process(
+            target=start_server,
+            args=(fs, host, self.server_port),
+        )
+        self.proc.start()
+        # Wait for server to start
+        wait_retry_backoff(
+            lambda: (None, check_port_open(host, self.server_port)),
+            timeout_secs=10,
+        )
+        return "grpc+tcp://{}:{}".format(host, self.server_port)
+
+    def create_offline_store_config(self) -> FeastConfigBaseModel:
+        self.remote_offline_store_config = RemoteOfflineStoreConfig(
+            type="remote", host="0.0.0.0", port=self.server_port
+        )
+        return self.remote_offline_store_config
+
+    def teardown(self):
+        super().teardown()
+        if self.proc is None and self.proc.is_alive():
```

Comment:
Shouldn't this be `if self.proc is not None and self.proc.is_alive():` ?
@dmartinol

Reply:
U r right. If you have time, coild you pls fix it? Otherwise I’ll do it tomorrow

Reply:
Will fix it

Reply:
seems like it is not yet fixed. just adding comment to avoid any confusion.

Reply:
also subprocess is not having `is_alive()` method just `is none` check enough.

Reply:
Thanks for catching will update in next pr